test/test_unistd.rs

```diff
@@ -104,6 +104,60 @@ mod linux_android {
     }
 }
 
+#[test]
+// `getgroups()` and `setgroups()` do not behave as expected on Apple platforms
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+fn test_setgroups() {
```

Comment:
These two tests race against each other.  You must add a mutex to prevent that.  grep for `CWD_MTX` for an example.